### DISCUSSION TITLE
Alarm comments

### DISCUSSION THREAD

**Discussion by maungawahau** [ottowayi/pycomm3#226] [Q&A]
Hi,

I've read the previous post on alarms on v31+ controllers which is seemingly unrelated to my question.
I'm using CompactLogix firmware v20.19 and I'd like to know if pycomm3 can read the comment associated with an alarm tag (of type ALARM_DIGITAL or ALARM_ANALOG). It seems the comment (the message that would be displayed in an alarm table in an HMI subscribed to the PLC) is not part of the ALARM_DIGITAL (or _ANALOG) tag structure. I presume this is because the alarm subscription method used by FactortTalk View SE supports natural languages, so many language specific  (English, German, etc) comments can be entered into the PLC so as to support the HMI user's (selected) language preference. I'm almost certain that the comments are stored in the PLC as FT View SE subscriber to the PLC and must pull the comments up from the device when an alarm is triggered. The comments are not predefined in the HMI as they would be in FT View Machine Edition (ME).
A tag and comment CSV export from RSLogix does expose comments associated with an alarm tag but they seem to be just that, a (language specific) comment and not tags themselves. The association with the alarm tag seems to come from having the same NAME but the comment has no DATATYPE. 
I'm inclined to think these are not easily addressable but I ask the question in hope that pycomm3 may in fact be able to return the comment associated with an alarm tag.

Cheers,
Simon

![image](https://user-images.githubusercontent.com/81834047/184809316-9149bbb9-6037-4bf1-aa72-fef004ae5bf5.png)

**Comment by ottowayi** [comment]
Those messages are stored differently and are not attributes of the tag. It may be possible to reverse engineer it, but right now it's not a feature in pycomm3.
